### PR TITLE
Add maven shell script formatting support via shfmt

### PR DESCRIPTION
**Pull request by tcrawford-figure**
This adds Maven support for shfmt and addresses https://github.com/diffplug/spotless/issues/1567. This is a follow up to https://github.com/diffplug/spotless/pull/1994. Closes https://github.com/diffplug/spotless/issues/1567 https://github.com/diffplug/spotless/issues/1776

### CODE REVIEW

File: plugin-maven/src/test/java/com/diffplug/spotless/maven/shell/ShellTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package com.diffplug.spotless.maven.shell;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.diffplug.spotless.maven.MavenIntegrationHarness;
+
+@Disabled // Disabled due to GHA not having shfmt
```
Comment (tcrawford-figure) [inline plugin-maven/src/test/java/com/diffplug/spotless/maven/shell/ShellTest.java:25]: Please let me know if I missed some detail that enables these types of formatters (ones that need to shell out) to run in GitHub action workflows. I'd be happy to add whatever is needed to get this to run in GHA.

**Review by Goooler — APPROVED** [review]

File: plugin-maven/src/test/java/com/diffplug/spotless/maven/shell/ShellTest.java (hunk 1 above)
Comment (nedtwigg) [reply plugin-maven/src/test/java/com/diffplug/spotless/maven/shell/ShellTest.java]: I replaced `@Disabled` with `@ShfmtTest`. If you run `gradlew test` these tests will not run. They only run when you run `./gradlew testShfmt` to run them.

- https://github.com/diffplug/spotless/blob/main/gradle/special-tests.gradle

If you want them to run on CI, you should add them with the npm test

https://github.com/diffplug/spotless/blob/797a665465429729b5308ee77f392c170c57242e/.github/workflows/ci.yml#L79-L81

- change it to `./gradlew testNpm testShfmt`

The only tricky part is getting shfmt onto the CI runner. Not that hard though. Feel free to open a PR for it and mess around.

Comment (tcrawford-figure) [reply plugin-maven/src/test/java/com/diffplug/spotless/maven/shell/ShellTest.java]: Perfect! Thank you for the clarification.

**Comment by nedtwigg** [comment]
Published in `plugin-maven 2.43.0`.

**Pull request merged**